platform reset API rename

alzix opened the pull request:
### Description
Renamed newly added `psa_system_reset()` API to `mbed_system_reset()`
We should not use PSA prefix for APIs not defined in PSA specs.

Must be merged after #9910 

Targeting RC2

<!--
    Required
    Add here detailed changes summary, testing results, dependencies
    Good example: https://os.mbed.com/docs/mbed-os/latest/contributing/workflow.html (Pull request template)
-->


### Pull request type

<!--
    Required
    Please add only one X to one of the following types. Do not fill multiple types (split the pull request otherwise).
    Please note this is not a GitHub task list, indenting the boxes or changing the format to add a '.' or '*' in front
    of them would change the meaning incorrectly. The only changes to be made are to add a description text under the
    description heading and to add a 'x' to the correct box.
-->
    [ ] Fix
    [X] Refactor
    [ ] Target update
    [ ] Functionality change
    [ ] Docs update
    [ ] Test update
    [ ] Breaking change

### Reviewers
@mikisch81 @orenc17 @mmahadevan108
<!--
    Optional
    Request additional reviewers with @username
-->

### Release Notes

<!--
    Optional
    In case of breaking changes, functionality changes or refactors, please add release notes here. 
    For more information, please see [the contributing guidelines](https://os.mbed.com/docs/mbed-os/latest/contributing/workflow.html#pull-request-types).
-->

mikisch81 commented on components/TARGET_PSA/inc/psa/lifecycle.h:
```diff
@@ -63,9 +64,11 @@ psa_status_t mbed_psa_reboot_and_request_new_security_state(uint32_t new_state);
 /** \brief Resets the system
- *
+ * 
+ * PSA targets do not allow NPS to access system power domain. 
```
NPS ->NSPE

mikisch81 commented on components/TARGET_PSA/inc/psa/lifecycle.h:
```diff
@@ -63,9 +64,11 @@ psa_status_t mbed_psa_reboot_and_request_new_security_state(uint32_t new_state);
- *
+ * 
+ * PSA targets do not allow NPS to access system power domain. 
+ * This API requests system reset to be caried out by SPE once all critical secure tasks are finished.
```
carried

orenc17 reviewed (changes requested):
please align with psa/errors.h

orenc17 commented on components/TARGET_PSA/inc/psa/client.h:
```diff
@@ -37,7 +37,6 @@
 
 #define PSA_FRAMEWORK_VERSION   (0x0100) /**< Version of the PSA Framework API. */
 #define PSA_VERSION_NONE        (0L)     /**< Identifier for an unimplemented Root of Trust (RoT) Service. */
-#define PSA_SUCCESS             (0L) /**< A general result code for calls to psa_call()  indicating success.*/
```
if you remove this add `#include psa/errors.h`

alzix replied:
i did not observe any compilation warning - but added anyway

alzix committed:
Rename psa_system_reset to mbed_psa_system_reset

add noreturn attributes
update lifecycle service to use psa/error.h
fix doxygen

0xc0170 reviewed (commented):
why `mbed_system_reset` is not in platform, and accessible to all ? The name leads me to believe this is common API.

0xc0170 reviewed (changes requested):
Line 57 - todo in lifecycle needs to be removed

I've checked lifecycle - 3 functions, 3 different prefixes

`psa_...`
`mbed_psa...`
`mbed_...`

does each one has special meaning ? why not all mbed_psa or just psa?
